### ISSUE TITLE
Checkboxes with self-referencing aria-labelledby are unreachable on Chrome

### ISSUE THREAD

**Issue by smhigley** [nvaccess/nvda#12138]
A checkbox with both of the following characteristics is unreachable in browse mode, using either arrow keys or form field shortcuts in Chrome (tested on v89):

- The checkbox uses `role="checkbox"`, and is unfocusable
- The checkbox has both an `aria-label` and self-referencing `aria-labelledby`

This only happens in Chrome (and Chromium Edge), not Firefox. The use case where this came up is a checkbox within an interactive grid, where it was the grid cell that actually took keyboard focus, but the checkbox itself should still work in browse mode.

I realize this is a bit of an edge case, and ideally the checkbox should probably be the one to take focus in this example, but I wonder if there are other use cases and implementations where the same issue might come up.

### Steps to reproduce:

Use the following code, or this [JS Fiddle](https://jsfiddle.net/ufjrmszL/5/show):

```html
  <label id="label1">First example</label>
  <input type="checkbox" id="check1" aria-label="native input" aria-labelledby="check1 label1">

  <label id="label2">Second example</label>
  <div role="checkbox" id="check2" aria-label="ARIA role, focusable" aria-labelledby="check2 label2" aria-checked="true" tabindex="0">✔️</div>

  <label id="label3">Third example</label>
  <div role="checkbox" id="check3" aria-label="ARIA role, not focusable" aria-labelledby="check3 label3" aria-checked="true">✔️</div>

  <label>Fourth example</label>
  <div role="checkbox" id="check3" aria-label="Fourth example, ARIA role, only aria-label, not focusable" aria-checked="true">✔️</div>
```

### Actual behavior:

The third checkbox is not reachable in browse mode, though all other checkboxes are.

### Expected behavior:

All checkboxes should be reachable.

#### Name and version of other software in use when reproducing the issue:
Chrome 89

Thanks!

**Comment by Adriani90** [comment]
@smhigley could you please open a bug in Chromium on crbug.com for this issue? You can reference this one there so we can keep track here.

**Comment by Adriani90** [comment]
Created https://bugs.chromium.org/p/chromium/issues/detail?id=1523736

Note that Jaws and Narator do not have this issue, it seems only related to NVDA.

cc: @michaelDCurran

**Comment by Adriani90** [comment]
This is not reproducible anymore in Chrome canary 133 or Edge 132. with NVDA 2024.4.1. Closing as works for me. If you are still having this issue, please comment and we can reopen.
